packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -18,26 +18,27 @@ class VisualStudio {
   /// Versions older than 2017 Update 2 won't be detected, so error messages to
   /// users should take into account that [false] may mean that the user may
   /// have an old version rather than no installation at all.
-  bool get isInstalled => _bestVisualStudioDetails != null;
+  bool get isInstalled => _bestVisualStudioDetails != null && _bestVisualStudioDetails.isNotEmpty;
 
   /// True if there is a version of Visual Studio with all the components
   /// necessary to build the project.
-  bool get hasNecessaryComponents => _usableVisualStudioDetails != null;
+  bool get hasNecessaryComponents => _usableVisualStudioDetails != null && _usableVisualStudioDetails.isNotEmpty;
 
   /// The name of the Visual Studio install.
   ///
   /// For instance: "Visual Studio Community 2017".
   String get displayName => _bestVisualStudioDetails[_displayNameKey];
 
+  /// The internal catalog map inside the installation details.
+  Map<String, dynamic> get _catalog => _bestVisualStudioDetails[_catalogKey];
```

Comment:
Optional: you could simplify the call sites using this by having it return an empty map instead of null; then they don't need to do null checks.